wandb/sdk/launch/agent/agent.py

```diff
@@ -182,23 +181,38 @@ def loop(self) -> None:
             while True:
                 self._ticks += 1
                 job = None
-                for queue in self._queues:
-                    job = self.pop_from_queue(queue)
-                    if job:
-                        break
-                if not job:
-                    time.sleep(AGENT_POLLING_INTERVAL)
-                    for job_id in self.job_ids:
-                        self._update_finished(job_id)
-                    if self._ticks % 2 == 0:
+                if self._running < self._max_jobs:
```

Comment:
did some fairly major restructuring here to handle the new max_jobs requirement, would be good to take a look

Reply:
So we only check the queues if we're not running the max jobs. Otherwise we just heartbeat (get_launch_agent for now), I think this is better.